internal/config/config.go

```diff
@@ -58,14 +59,17 @@ type Config struct {
 	Version string `yaml:"version" validate:"eq=v2"`
 	// description: |
 	//   Machine image used to create Constellation nodes.
-	Image string `yaml:"image" validate:"required"`
+	Image string `yaml:"image" validate:"required,version_compatibility"`
 	// description: |
 	//   Size (in GB) of a node's disk to store the non-volatile state.
 	StateDiskSizeGB int `yaml:"stateDiskSizeGB" validate:"min=0"`
 	// description: |
-	//   Kubernetes version to be installed in the cluster.
+	//   Kubernetes version to be installed into the cluster.
 	KubernetesVersion string `yaml:"kubernetesVersion" validate:"supported_k8s_version"`
```

Comment:
Would it make sense to have a `Version` type here (and for microservice version etc..)? We have a lot of code now that accesses major, minor, patch, .. fields.
We could also have consistent de-/encoding with or without the `v` prefix.
This could make our version handling more consistent and maybe help with some of the feedback from @daniel-weisse?

Reply:
I guess this Version lib could make sense in streamlining version handling: https://pkg.go.dev/github.com/hashicorp/go-version

Reply:
Yeah looks like this could work great for us. 
Since this PR already has a lot of changes we can also move this to a seperate one.